grails-app/services/au/org/ala/bie/SearchService.groovy

```diff
@@ -887,12 +887,41 @@ class SearchService {
         parser.parseText(response)
     }
 
+    def split(xs, batchSize) {
+        def batches = []
+        def currentBatch = []
+
+        xs.each { x ->
+            currentBatch << x
+
+            if(currentBatch.size() == batchSize) {
+                batches << currentBatch
+                currentBatch = []
+            }
+        }
+
+        if(!currentBatch.isEmpty()) {
+            batches << currentBatch
```

Comment:
what would happen if you'd push empty currentBatch into batches?

Reply:
In this case, not much - just an extra request to biocache-service